mediagis/nominatim-docker
# Change import style

Nominatim allows several import styles to [filter imported data ](https://nominatim.org/release-docs/3.7.0/admin/Import/#filtering-imported-data)

I tried to implement the proposition in  https://github.com/mediagis/nominatim-docker/issues/173.
It does not work as expected and I would like some help to make it right.
Thanks in advance.

I made it as a pull request as I think it can interest others too.

How to reproduce:

1. Create nominatim instance with new parameter :
```bash
docker run -it --rm \
  -e PBF_URL=https://download.geofabrik.de/europe/luxembourg-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/luxembourg-updates/ \
  -e IMPORT_WIKIPEDIA=false \
  -e NOMINATIM_PASSWORD=very_secure_password \
  -e THREADS=6 \
  -e IMPORT_STYLE=admin \
  -p 8080:8080 \
  --name nominatim-luxembourg-admin \
  mediagis/nominatim:3.6
```

2. Check size
- connect to database
```bash
docker exec -i -t nominatim /bin/bash
su - postgres
psql -d nominatim
```
- send SQL query

```sql
nominatim=# select schemaname as table_schema, relname as table_name, pg_size_pretty(pg_relation_size(relid)) as data_size from pg_catalog.pg_statio_user_tables order by pg_relation_size(relid) desc;
 table_schema |        table_name         | data_size
--------------+---------------------------+------------
 public       | planet_osm_nodes          | 136 MB
 public       | planet_osm_ways           | 100 MB
 public       | placex                    | 97 MB
 public       | place                     | 49 MB
 public       | search_name               | 23 MB
 public       | country_osm_grid          | 16 MB
 public       | place_addressline         | 7904 kB
 public       | search_name_74            | 7424 kB
 public       | spatial_ref_sys           | 6936 kB
 public       | location_road_74          | 5664 kB
 public       | word                      | 5568 kB
 public       | planet_osm_rels           | 3400 kB
 public       | location_area_large_74    | 848 kB
 public       | location_postcode         | 552 kB
 public       | location_road_15          | 192 kB
 public       | search_name_4             | 192 kB
 public       | location_road_3           | 176 kB
 public       | search_name_3             | 152 kB
 public       | country_name              | 88 kB
 public       | search_name_15            | 80 kB
 public       | location_area_country     | 56 kB
 public       | location_area_large_4     | 56 kB
 public       | location_road_4           | 56 kB
 public       | location_area_large_3     | 16 kB
 public       | import_status             | 8192 bytes
 public       | address_levels            | 8192 bytes
 public       | location_area_large_15    | 8192 bytes
``` 
3. Create without new parameter

```bash
docker run -it --rm \
  -e PBF_URL=https://download.geofabrik.de/europe/luxembourg-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/luxembourg-updates/ \
  -e IMPORT_WIKIPEDIA=false \
  -e NOMINATIM_PASSWORD=very_secure_password \
  -e THREADS=6 \
  -p 8080:8080 \
  --name nominatim-luxembourg-admin \
  mediagis/nominatim:3.6
```
4. Compare size

```sql
nominatim=# select schemaname as table_schema, relname as table_name, pg_size_pretty(pg_relation_size(relid)) as data_size from pg_catalog.pg_statio_user_tables order by pg_relation_size(relid) desc;
 table_schema |        table_name         | data_size
--------------+---------------------------+------------
 public       | planet_osm_nodes          | 136 MB
 public       | planet_osm_ways           | 100 MB
 public       | placex                    | 97 MB
 public       | place                     | 49 MB
 public       | search_name               | 23 MB
 public       | country_osm_grid          | 16 MB
 public       | place_addressline         | 7904 kB
 public       | search_name_74            | 7424 kB
 public       | spatial_ref_sys           | 6936 kB
 public       | location_road_74          | 5664 kB
 public       | word                      | 5568 kB
 public       | planet_osm_rels           | 3400 kB
 public       | location_area_large_74    | 848 kB
 public       | location_postcode         | 552 kB
 public       | location_road_15          | 192 kB
 public       | search_name_4             | 192 kB
 public       | location_road_3           | 176 kB
 public       | search_name_3             | 152 kB
 public       | country_name              | 88 kB
 public       | search_name_15            | 80 kB
 public       | location_area_country     | 56 kB
 public       | location_area_large_4     | 56 kB
 public       | location_road_4           | 56 kB
 public       | location_area_large_3     | 16 kB
 public       | import_status             | 8192 bytes
 public       | address_levels            | 8192 bytes
 public       | location_area_large_15    | 8192 bytes
``` 
As you can see, the numbers are the same, or with the parameter, the database should be lighter. 

It might be an easiest way to compare the 2 databases but I didn't thought of one.
